kernel/cemitcodes.ml

```diff
@@ -369,10 +369,17 @@ let to_memory (init_code, fun_code, fv) =
   init();
   emit init_code;
   emit fun_code;
-  let code = String.create !out_position in
-  String.unsafe_blit !out_buffer 0 code 0 !out_position;
   (** Later uses of this string are all purely functional *)
-  let code = CString.hcons code in
+  let code = Bytes.init !out_position (fun i ->
+      Bytes.unsafe_get !out_buffer i
+    ) in
```

Comment:
You can use `let code = Bytes.sub_string !out_buffer 0 !out_position` to get a string (or `Bytes.sub` to get a bytes) safely and efficiently -- it corresponds to the `unsafe_blit` version that was there before.